src/main/java/io/gravitee/policy/cache/CachePolicy.java

```diff
@@ -51,20 +51,22 @@
 
     private final static char KEY_SEPARATOR = '_';
 
-    private final static String BYPASS_CACHE_QUERY_PARAMETER = "bypass-cache";
-    private final static String X_GRAVITEE_BYPASS_CACHE = "X-Gravitee-Bypass-Cache";
+    // Policy cache action
+    private final static String CACHE_ACTION_QUERY_PARAMETER = "cache";
+    private final static String X_GRAVITEE_CACHE_ACTION = "X-Gravitee-Cache";
 
-    private volatile Cache cache;
+    private Cache cache;
+    private CacheAction action;
 
     public CachePolicy(final CachePolicyConfiguration cachePolicyConfiguration) {
         this.cachePolicyConfiguration = cachePolicyConfiguration;
     }
 
     @OnRequest
     public void onRequest(Request request, Response response, ExecutionContext executionContext, PolicyChain policyChain) {
-        boolean bypassCache = lookForByPass(request);
+        action = lookForAction(request);
 
-        if (! bypassCache) {
+        if (action == null || action != CacheAction.BY_PASS) {
```

Comment:
You don't have to test if `action` is null. You could simply test `action != CacheAction.BY_PASS`